tests/test_forms.py

```diff
@@ -348,3 +348,50 @@ def test_widgets_selected_after_validation_error(self, db, live_server, driver):
         # genres should still have One as selected option
         result_title = driver.find_element_by_css_selector('.select2-selection--multiple li').get_attribute('title')
         assert result_title == 'One'
+
+
+class TestAddressChainedSelect2Widget(object):
+    url = reverse('model_chained_select2_widget')
+    form = forms.AddressChainedSelect2WidgetForm()
+
+    def test_widgets_selected_after_validation_error(self, db, live_server, driver, countries, cities):
+        driver.get(live_server + self.url)
+
+        country_container, city_container = driver.find_elements_by_css_selector('.select2-selection--single')
+
+        # clicking city select2 lists all available cities
+        city_container.click()
+        city_options = driver.find_elements_by_css_selector('.select2-results li')
+        city_names_from_browser = set([option.text for option in city_options])
```

Comment:
you don't need to `set([])` you can cast a generator directly into a set or write a set generator:
```python
{option.text for option in city_options}
```